### ISSUE TITLE
Use organism names in metagenotype display name

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#2156]
@CuzickA has noticed that the current metagenotype display name isn't useful when you have a wild-type host with no genes. All you get is the text 'host: wild type', which tells you nothing about what the host species is:

![image](https://user-images.githubusercontent.com/37659591/70537700-d0a88b00-1b58-11ea-97db-c717b60b05ff.png)

This display name is also inconsistent with how we display the metagenotype name in the annotation edit modal:

![image](https://user-images.githubusercontent.com/37659591/70537800-f7ff5800-1b58-11ea-958d-89b8cb8d2593.png)

...and in the annotation tables on the summary page:

![image](https://user-images.githubusercontent.com/37659591/70537911-211fe880-1b59-11ea-8ed9-449f13fee0b8.png)

We should amend the default display name to include more information (at least the organism name, though we probably need the strain name too).

**Comment by jseager7** [comment]
I've looked into this, but I'm not sure there's much I can do because the `display_name` is generated on the back-end.

Specifically, it looks like `lib/Canto/CursDB/Metagenotype.pm` doesn't have any way to get the organism name or the strain name. For example, you can't do `$self->pathogen_genotype->organism->full_name()`. This isn't the same as using the `Metagenotype` service on the front-end; that has all of the organisms and strain names joined to the host and pathogen genotypes.

@kimrutherford how should we proceed? Should the back-end include organism and strain information when returning a metagenotype, or should we try to migrate more of these templates (`title.mhtml`, `metagenotype_page.mhtml`, and so on) to AngularJS, so we can use the AngularJS services instead?

**Comment by kimrutherford** [comment]
> Should the back-end include organism and strain information when returning a metagenotype,
> or should we try to migrate more of these templates (title.mhtml, metagenotype_page.mhtml,
> and so on) to AngularJS, so we can use the AngularJS services instead?

I think both of those make sense but probably returning more information from the back-end is the quicker fix.  Let's chat about this next time on Skype.

**Comment by kimrutherford** [comment]
> I think both of those make sense but probably returning more information from the back-end
> is the quicker fix.

Now I've checked, I think this information is already available.  The metagenotype list service returns separate "host_genotype" and "pathogen_genotype" fields which could be used to create a display name in the JS code.  Here's a truncated JSON example from the service:

```json
[
  {
    "feature_id": 1,
    "display_name": "POLG_TRMVUdelta Triticum mosaic virus (isolate Triticum aestivum/United States/U06-123/2006) (Unknown strain) / wild type Solanum tuberosum (Unknown strain)",
    "annotation_count": 1,
    "metagenotype_id": 1,
    "host_genotype": {
      "name": null,
      "strain_name": "Unknown strain",
      "organism": {
        "pathogen_or_host": "host",
        "taxonid": "4113",
        "common_name": "Potato",
        "scientific_name": "Solanum tuberosum",
        "genotype_count": 1,
        "full_name": "Solanum tuberosum",

```

**Comment by jseager7** [comment]
> The metagenotype list service returns separate "host_genotype" and "pathogen_genotype" fields which could be used to create a display name in the JS code.

Is this the `Metagenotype` AngularJS service? I could use that, but it's a bit awkward at the moment since both templates use Mason. I could just drop in a directive that renders the name properly, but I'll still need a way to fetch the correct metagenotype from the `Metagenotype` service (because the service behaves like a model service, in that it fetches all the metagenotypes rather than querying for one of them). If the back-end is providing some ID for the selected metagenotype, then this shouldn't be a problem.

**Comment by kimrutherford** [comment]
> both of the templates that need this change are using Mason

Which templates need to change.  I thought it was the edit dialog where the change was needed?

**Comment by jseager7** [comment]
> Which templates need to change. I thought it was the edit dialog where the change was needed?

It's actually the templates `title.mhtml` and `metagenotype_page.mhtml`. My original post is a bit misleading because I was presenting the image of the edit modal as an example of how the metagenotype display name property (the one provided by the backend) is _inconsistent_ with what is shown in the edit modal (because the display name set by the backend doesn't include organism names).

This is the display used in `title.mhtml` that needs changing to include organism names:

![image](https://user-images.githubusercontent.com/37659591/70537700-d0a88b00-1b58-11ea-97db-c717b60b05ff.png)

There's also this section of `metagenotype_page.mhtml` that uses the same format:

![image](https://user-images.githubusercontent.com/37659591/71887712-90780080-3136-11ea-90ef-bcf5ff5c3207.png)

**Comment by kimrutherford** [comment]
Hi James.  I've created a PR with this change: pombase/canto#2193

Could you try it out when you get a chance?  I'm happy to tweak things before we merge it.

**Comment by jseager7** [comment]
Closed by #2193.
